### DISCUSSION TITLE
Unit testing, writing a RouterContext

### DISCUSSION THREAD

**Discussion by AndyDecker** [oakserver/oak#324] [Q&A]
Hello,
i recently started my first deno server as a uni project and now i'm trying to write unit tests for my application. Since i normally pass down the RouterContext to my functions i wondered if there is an easy way (or an already written down example) for a RouterContext to use as a parameter for my functions under test.
Thanks for any help!
Best regards,
Andreas

**Comment by kitsonk** [comment]
I need to formalise it a bit better and make some of the testing mocks part of the formal public API, but https://github.com/oakserver/oak/blob/main/router_test.ts gives a pretty good example of how you can do this.
